client/service/src/builder.rs

```diff
@@ -893,6 +919,26 @@ where
 		.extra_sets
 		.insert(0, transactions_handler_proto.set_config());
 
+	// crate statment protocol and add it to the list of supported protocols of `network_params`
+
+	let statement_handler_proto = if statement_store.is_some() {
+		let statement_handler_proto = sc_network_statement::StatementHandlerPrototype::new(
+			protocol_id.clone(),
+			client
+				.block_hash(0u32.into())
+				.ok()
+				.flatten()
+				.expect("Genesis block exists; qed"),
+			config.chain_spec.fork_id(),
+		);
+		network_params
+			.network_config
+			.extra_sets
+			.insert(0, statement_handler_proto.set_config());
+		Some(statement_handler_proto)
+	} else {
+		None
+	};
```

Comment:
This should be passed in `extra_sets`. We don't want to change this functionality here anymore because that "forces" people to use these things.

Reply:
What do you mean? It is inserted in `extra_sets`. The protocol handler is an `Option` so it does not force usage.

Reply:
I mean that this doesn't need to be a depedency in `sc-service. We should get rid off the "include all the things in sc-service". There is no need to have this as an option here.

Reply:
Done